File: src/main/java/ui/pageobjects/fragments/searchresults/VideoResultItem.java
package ui.pageobjects.fragments.searchresults;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import ui.pageobjects.pages.VideoPlayerPage;

public class VideoResultItem {
    private final Page page;
    private final Locator wrapper;

    public VideoResultItem(Page page, Locator wrapper) {
        this.page = page;
        this.wrapper = wrapper;
    }

    public Locator thumbnailImage() {
        return this.getWrapper().locator(".iris_thumbnail img");
    }

    public Locator title() {
        return this.getWrapper().locator(".iris_video-vital__title");
    }

    public Locator userInfo() {
        return this.getWrapper().locator(".iris_userinfo");
    }

    public Locator getWrapper() {
        return this.wrapper;
    }

    public VideoPlayerPage clickThumbnail() {
        this.thumbnailImage().click();
        return new VideoPlayerPage(this.page);
    }

    public VideoPlayerPage clickTitle() {
        this.title().click();
        return new VideoPlayerPage(this.page);
    }
}

File: src/main/kotlin/ui/pageobjects/fragments/searchresults/VideoResultItem.kt
package ui.pageobjects.fragments.searchresults

import com.microsoft.playwright.Locator
import com.microsoft.playwright.Page
import ui.pageobjects.pages.VideoPlayerPage

class VideoResultItem(private val page: Page, private val wrapper: Locator) {
    fun thumbnailImage(): Locator {
        return wrapper.locator(".iris_thumbnail img")
    }

    fun title(): Locator {
        return wrapper.locator(".iris_video-vital__title")
    }

    fun userInfo(): Locator {
        return wrapper.locator(".iris_userinfo")
    }

    fun clickThumbnail(): VideoPlayerPage {
        thumbnailImage().click()
        return VideoPlayerPage(this.page)
    }

    fun clickTitle(): VideoPlayerPage {
        title().click()
        return VideoPlayerPage(this.page)
    }
}
